app/controllers/admin/themes_controller.rb

```diff
@@ -280,4 +282,10 @@ def log_theme_change(old_record, new_record)
     StaffActionLogger.new(current_user).log_theme_change(old_record, new_record)
   end
 
+  def handle_switch
+    param = theme_params[:component]
+    if ["true", "false"].include?(param.to_s) && param.to_s != @theme.component.to_s
+      @theme.switch_type!
```

Comment:
I don't like this API, toggle is not great, it should explicitly either set something to a component or a theme. 

@theme.component = params.to_s == "true"
@theme.save! 

something like that ...